src/lcm/grid_helpers.py

```diff
@@ -0,0 +1,111 @@
+"""Functions to generate and work with different kinds of grids.
+
+Grid generation functions must have the following signature:
+
+    Signature (start: Scalar, stop: Scalar, n_points: int) -> jax.Array
+
+They take start and end points and create a grid of points between them.
+
+
+Interpolation info functions must have the following signature:
+
+    Signature (
+        value: Scalar,
+        start: Scalar,
+        stop: Scalar,
+        n_points: int
+    ) -> Scalar
+
+They take the information required to generate a grid, and return an index corresponding
+to the value, which is a point in the space but not necessarily a grid point.
+
+Some of the arguments will not be used by all functions but the aligned interface makes
+it easy to call functions interchangeably.
+
+"""
+
+import jax.numpy as jnp
+from jax import Array
+
+from lcm.typing import Scalar
+
+
+def linspace(start: Scalar, stop: Scalar, n_points: int) -> Array:
+    """Wrapper around jnp.linspace.
+
+    Returns a linearly spaced grid between start and stop with n_points, including both
+    endpoints.
+
+    """
+    return jnp.linspace(start, stop, n_points)
+
+
+def get_linspace_coordinate(
+    value: Scalar,
+    start: Scalar,
+    stop: Scalar,
+    n_points: int,
```

Comment:
I think it would make sense to have `get_..._coordinate` and the creation function as methods of the grid object. 

If not, I would at least replace these three arguments by a `grid` or `gridspec` argument. 

(Implementation can of course stay in these functions but they would become private)

Reply:
With this we could get to a point where a user could add a new grid type without changing LCM source code.